src/main/java/com/cloudbees/jenkins/support/configfiles/SecretHandler.java

```diff
@@ -87,4 +91,8 @@ public void characters(char[] ch, int start, int length) throws SAXException {
 
         return patchedFile;
     }
+
+    private static boolean isSecretBytes(String value) {
+        return (SecretBytes.decrypt(Base64.decode(value.substring(1, value.length() - 1))) != null);
```

Comment:
Will have issues if the string is length 1.

You could check first that the string starts with `{` ends with `}` and is at least 8 characters long (may be able to increase the min length further if we made assumptions about the salt length)